### PR TITLE
fixes #21970 - remove broken config validation check in scratch mode

### LINKED ISSUE #21970: teleport-cluster deployment fails in scratch mode with helm charts v12.0.2
Current behavior:
teleport-cluster-auth-test and teleport-cluster-proxy-test jobs failed in "scratch mode" with helm chart v12.0.2 with this error :
FAIL /etc/teleport/teleport.yaml
ERROR: failed parsing the config file: yaml: unmarshal errors:  line 1: field Error not found in type config.FileConfig

After some investigation, it's caused by 2 erroneous lines into _config.scratch.tpl files like: 
{{- required "'auth.teleportConfig' is required in scratch mode" .Values.auth.teleportConfig }}
and
{{- required "'proxy.teleportConfig' is required in scratch mode" .Values.proxy.teleportConfig }}

Bug details:
- Teleport version: 12.0.2
- Recreation steps: use scratch mode

### PR DESCRIPTION
**Pull request by anthonysomerset**
this fixes #21970

### CODE REVIEW

**Comment by PReimers** [comment]
I might be wrong, but is there a reason you reference the proxy configuration using `$proxy.proxy.teleportConfig`?
In the auth configuration, you used `$auth.teleportConfig` (instead of `$auth.auth.teleportConfig`)

**Comment by anthonysomerset** [comment]
Just a typo on my part. Will fix


On 22 Feb 2023, at 16:55, Patrick Reimers ***@***.***> wrote:

﻿

I might be wrong, but is there a reason you reference the proxy configuration using $proxy.proxy.teleportConfig?
In the auth configuration, you used $auth.teleportConfig (instead of $auth.auth.teleportConfig)

**Comment by anthonysomerset** [comment]
yes - auth was correct - proxy accidentally had a double reference which is now fixed

**Comment by PReimers** [comment]
@jimbishopp @klizhentas @r0mant @russjones @zmb3

Sorry for the ping.
Any chance this could get merged (and released) soon? 🥺

It's currently the only thing preventing us from using teleport (we can't deploy it 🙈).

**Comment by anthonysomerset** [comment]
@PReimers  if you know your config is good - or you wish to go through the pain of debugging and do uninstall/redeploy's etc

you can add this line into your values file:

validateConfigOnDeploy: false

**Review by hugoShaka — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

File: examples/chart/teleport-cluster/templates/proxy/predeploy_config.yaml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {{- $proxy := mustMergeOverwrite (mustDeepCopy .Values) .Values.proxy -}}
 {{- if $proxy.validateConfigOnDeploy }}
+{{- if not $proxy.proxy.teleportConfig -}}
+{{- fail "'proxy.teleportConfig' is required in scratch mode" }}
+{{- end -}}
```
Comment (hugoShaka) [inline examples/chart/teleport-cluster/templates/proxy/predeploy_config.yaml:5]: idem
```suggestion
```

File: examples/chart/teleport-cluster/templates/proxy/predeploy_config.yaml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 {{- $proxy := mustMergeOverwrite (mustDeepCopy .Values) .Values.proxy -}}
 {{- if $proxy.validateConfigOnDeploy }}
+{{- if not $proxy.teleportConfig -}}
+{{- fail "'proxy.teleportConfig' is required in scratch mode" }}
+{{- end -}}
```
Comment (hugoShaka) [inline examples/chart/teleport-cluster/templates/proxy/predeploy_config.yaml:5]: idem

```suggestion
```

**Comment by hugoShaka** [comment]
@anthonysomerset thank you for the contribution 🎉 . I dropped the check entirely as it added little value and users switching to scratch mode are already power users and such validations are more harmful than useful for them.

**Pull request merged**